### PR TITLE
Refactored Habit Statistic Controller Tests

### PR DESCRIPTION
**Pull request by greatpanda1603**

### CODE REVIEW

File: core/src/test/java/greencity/controller/HabitStatisticControllerTest.java [hunk 1]
```diff
@@ -1,111 +1,114 @@
...
 import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.patch;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
 
-@SpringBootTest
-@AutoConfigureMockMvc
-@RunWith(SpringRunner.class)
+@ExtendWith(SpringExtension.class)
```
Comment (antonbakalets) [inline core/src/test/java/greencity/controller/HabitStatisticControllerTest.java:31]: `@ExtendWith(MockitoExtension.class)` should be enough as there is no Spring context here, only mocked dependencies.

File: core/src/test/java/greencity/controller/HabitStatisticControllerTest.java [hunk 2]
```diff
@@ -1,111 +1,114 @@
...
+    @InjectMocks
+    private HabitStatisticController habitStatisticController;
+
+    @Mock
+    private HabitStatisticService habitStatisticService;
+
+    @BeforeEach
+    void setup() {
+        this.mockMvc = MockMvcBuilders.standaloneSetup(habitStatisticController)
+            .alwaysDo(print())
```
Comment (antonbakalets) [inline core/src/test/java/greencity/controller/HabitStatisticControllerTest.java:45]: Should not be committed.

File: core/src/test/java/greencity/controller/HabitStatisticControllerTest.java (hunk 1 above)
Comment (greatpanda1603) [reply core/src/test/java/greencity/controller/HabitStatisticControllerTest.java]: Used Mockito extension, updated respective dependencies in POM

File: core/src/test/java/greencity/controller/HabitStatisticControllerTest.java (hunk 2 above)
Comment (greatpanda1603) [reply core/src/test/java/greencity/controller/HabitStatisticControllerTest.java]: Removed .alwaysDo(print())

**Pull request merged**